### PR TITLE
feat: declare maxCount in BaseScript

### PR DESCRIPTION
**Pull request by andreivladbrg**
I have named it `maxCount` because it is going to be used for both segments and tranches.

@PaulRBerg fyi i will update the warp drive

### CODE REVIEW

File: script/Base.s.sol [hunk 1]
```diff
@@ -22,6 +25,10 @@ contract BaseScript is Script {
     /// @dev Used to derive the broadcaster's address if $ETH_FROM is not defined.
     string internal mnemonic;
 
+    /// @dev The upper limit on the size of specific data structures that are placed in the contract storage to ensure
+    /// transactions stay within the block gas limit.
+    uint256 internal maxCount;
```
Comment (smol-ninja) [inline script/Base.s.sol:30]: should be above `mnemonic` (alphabetically)

**Review by smol-ninja — CHANGES_REQUESTED** [review]
LGTM.

File: script/Base.s.sol [hunk 2]
```diff
@@ -22,6 +25,10 @@ contract BaseScript is Script {
     /// @dev Used to derive the broadcaster's address if $ETH_FROM is not defined.
     string internal mnemonic;
 
+    /// @dev The upper limit on the size of specific data structures that are placed in the contract storage to ensure
+    /// transactions stay within the block gas limit.
```
Comment (PaulRBerg) [inline script/Base.s.sol:29]: Let's change to:

> The upper limit on the length of data structures to ensure that transactions stay within the block gas limit

File: script/Base.s.sol [hunk 3]
```diff
@@ -37,6 +44,13 @@ contract BaseScript is Script {
             mnemonic = vm.envOr({ name: "MNEMONIC", defaultValue: TEST_MNEMONIC });
             (broadcaster,) = deriveRememberKey({ mnemonic: mnemonic, index: 0 });
         }
+
+        // Sets `maxCount` to 300 for Avalanche, or 500 for all other chains.
```
Comment (PaulRBerg) [inline script/Base.s.sol:48]: `and` instead of `or`.

**Review by PaulRBerg — CHANGES_REQUESTED** [review]
feedback below

**Comment by andreivladbrg** [comment]
addressed your suggestions, does it look good now?

**Review by smol-ninja — APPROVED** [review]

**Pull request merged**